go/tasks/plugins/array/k8s/config.go

```diff
@@ -31,14 +37,80 @@ var (
 		},
 	}
 
-	configSection = config.MustRegisterSection(configSectionKey, defaultConfig)
+	configSection = pluginsConfig.MustRegisterSubSection(configSectionKey, defaultConfig)
 )
 
+type ResourceConfig struct {
+	PrimaryLabel string `json:"primaryLabel" pflag:",PrimaryLabel of a given service cluster"`
+	Limit        int    `json:"limit" pflag:",Resource quota (in the number of outstanding requests) for the cluster"`
+}
+
+type ClusterConfig struct {
+	Name     string `json:"name" pflag:",Friendly name of the remote cluster"`
+	Endpoint string `json:"endpoint" pflag:", Remote K8s cluster endpoint"`
+	Auth     Auth   `json:"auth" pflag:"-, Auth setting for the cluster"`
+	Enabled  bool   `json:"enabled" pflag:", Boolean flag to enable or disable"`
+}
+
+type Auth struct {
+	Type      string `json:"type" pflag:", Authentication type"`
+	TokenPath string `json:"tokenPath" pflag:", Token path"`
+	CertPath  string `json:"certPath" pflag:", Certificate path"`
+}
+
+func (auth Auth) GetCA() ([]byte, error) {
+	cert, err := ioutil.ReadFile(auth.CertPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read k8s CA cert from configured path")
+	}
+	return cert, nil
+}
+
+func (auth Auth) GetToken() (string, error) {
+	token, err := ioutil.ReadFile(auth.TokenPath)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to read k8s bearer token from configured path")
+	}
+	return string(token), nil
+}
+
+// TODO: Move logic to flytestdlib
+// Reads secret values from paths specified in the config to initialize a Kubernetes rest client Config.
+func RemoteClusterConfig(host string, auth Auth) (*restclient.Config, error) {
```

Comment:
What is the purpose of remoteCluster?

Reply:
This is our V0 milestone. We do not have a way to prioritize regular tasks over array tasks in the same namespace